### PR TITLE
factor out references to the android SDK target into an env variable

### LINKED ISSUE #8348: Reduce duplication of the "android-18" property in the Android build
If you look at:
https://github.com/servo/servo/search?utf8=%E2%9C%93&q=android-18

You'll see that we have copies of the string "android-18" in a project properties file, a .mk file, and some Rust code. This can probably be broken out into an environment variable set in the Mach/python driver that is expanded in the appropriate ways for the .mk file and Rust code, and gets generated into the properties file by passing the correct flags to the `android project update` command.

Instructions for building to target Android are available here:
https://github.com/servo/servo/wiki/Building-for-Android

### PR DESCRIPTION
**Pull request by ajnirp**
fixes #8348

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8519)

### CODE REVIEW

File: python/servo/post_build_commands.py [hunk 1]
```diff
@@ -182,6 +182,9 @@ def package(self, params, release=False, dev=False, debug=False, debugger=None):
         target_dir = path.join(self.get_target_dir(), "arm-linux-androideabi")
         dev_flag = ""
 
+        # Set the Android SDK version target
+        env['ANDROID_TARGET'] = 'android-18'
```
Comment (Manishearth) [inline python/servo/post_build_commands.py]: Should we make this a part of .servobuild instead?

**Comment by larsbergstrom** [comment]
@bors-servo r+

I think we can factor this out into a setting or argument later. If we end up producing builds for multiple android target versions (as Fennec does today!) we'll need to be able to set it as an argument on the build machines, which don't use .servobuild today.

**Comment by jdm** [comment]
```
make[2]: Entering directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k/crypto'
In file included from cryptlib.c:117:0:
cryptlib.h:62:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
mem.c:59:19: fatal error: stdio.h: No such file or directory
 #include <stdio.h>
                   ^
compilation terminated.
mem_clr.c:59:20: fatal error: string.h: No such file or directory
 #include <string.h>
                    ^
compilation terminated.
mem_dbg.c:112:19: fatal error: stdio.h: No such file or directory
 #include <stdio.h>
                   ^
compilation terminated.
In file included from cversion.c:59:0:
cryptlib.h:62:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
In file included from ex_data.c:141:0:
cryptlib.h:62:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
cpt_err.c:61:19: fatal error: stdio.h: No such file or directory
 #include <stdio.h>
                   ^
compilation terminated.
In file included from uid.c:56:0:
../include/openssl/crypto.h:120:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
o_time.c:63:20: fatal error: string.h: No such file or directory
 #include <string.h>
                    ^
compilation terminated.
o_str.c:59:19: fatal error: ctype.h: No such file or directory
 #include <ctype.h>
                   ^
compilation terminated.
o_dir.c:59:19: fatal error: errno.h: No such file or directory
 #include <errno.h>
                   ^
compilation terminated.
In file included from o_fips.c:58:0:
cryptlib.h:62:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
In file included from o_init.c:55:0:
../e_os.h:451:30: fatal error: unistd.h: No such file or directory
 #      include OPENSSL_UNISTD
                              ^
compilation terminated.
make[2]: *** [depend] Error 1
make[2]: Leaving directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k/crypto'
make[1]: *** [depend] Error 1
make[1]: Leaving directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k'
make[1]: Entering directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k'
making all in crypto...
make[2]: Entering directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k/crypto'
arm-linux-androideabi-gcc -I. -I.. -I../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -Wa,--noexecstack -march=armv7-a -mandroid -I/home/servo/android-ndk-r10c//platforms//arch-arm/usr/include -B/home/servo/android-ndk-r10c//platforms//arch-arm/usr/lib -O3 -fomit-frame-pointer -Wall -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DAES_ASM -DGHASH_ASM   -c -o cryptlib.o cryptlib.c
In file included from cryptlib.c:117:0:
cryptlib.h:62:20: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h>
                    ^
compilation terminated.
make[2]: *** [cryptlib.o] Error 1
make[2]: Leaving directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k/crypto'
make[1]: *** [build_crypto] Error 1
make[1]: Leaving directory `/home/servo/buildbot/slave/android/build/support/android/openssl-1.0.1k'
make: *** [openssl-1.0.1k/libssl.so] Error 2
```

**Comment by wenderen** [comment]
@jdm I'm unable to repro this. How do I do so?

**Comment by jdm** [comment]
Maybe @larsbergstrom has an idea?

**Comment by larsbergstrom** [comment]
I will try this locally and see if I can repro it! Sorry for the delay.

**Comment by larsbergstrom** [comment]
Very strange! `./mach build --android -d && ./mach package -d` worked for me, too. I'll have to try it on the builder.

**Comment by larsbergstrom** [comment]
This does not repro on the builder, either, with this branch and an identical environment. I'd like to give this a retry, in case it's something related to our often-flaky OpenSSL build process (we download and extract a tarball, and it frequently causes issues).

@bors-servo retry

**Comment by larsbergstrom** [comment]
Well, at least it's reproducible! I'll try to repro harder :-)

**Comment by larsbergstrom** [comment]
Ha! Managed to repro it. Need to `git clean -fdx` because the problem is that if you've already d/l'd and built openssl, even a `./mach clean` won't get rid of it. The directory is support/android/openssl-1.0.1k.

Let me know if you'd like me to help with tracking down the issue, and sorry it took me so long to figure out how to repro! I totally forgot that we build openssl in-tree. I should open issues for us to move our in-tree builds to the target directory...

**Comment by larsbergstrom** [comment]
@wenderen Do you need any assistance with this PR?

**Comment by jdm** [comment]
Closing due to lack of updates.

**Pull request closed without merging**